draft-ietf-dnsop-dns-catalog-zones.md

```diff
@@ -460,53 +458,40 @@ zones), but nameservers MUST NOT process such broken zones as catalog
 zones.  For the purpose of catalog processing, the broken catalogs MUST be
 ignored.
 
-[FIXME: reword this if we do coo, or the CSYNC immediate flag, or something else]
-If there is a clash between an existing member zone's name and an incoming
-member zone's name (via transfer or update), the new instance of the zone MUST
-be ignored and an error SHOULD be logged.
-
-## Member zone name clash
-
-## Member zone removal
-
-When zones are deleted from a catalog zone, a primary MAY delete the member
-zone immediately.
-It is up to the secondary nameserver to handle this condition correctly.
+## Member zone removal {#zoneremoval}
 
-## Migrating zones between catalogs
+When a member zone is removed from a specific catalog zone, an authoritative server MUST NOT remove the zone and associated state data if the zone was not configured from that specific catalog zone.
+Only when the zone was configured from a specific catalog zone, and the zone is removed as a member from that specific catalog zone, the zone and associated state (such as zone data and DNSSEC keys) MUST be removed.
```

Comment:
How do we handle the case where an operator introduces existing zones as member zones to a catalog (i.e., bootstrapping a catalog in the presence of existing zones)? Taking this sentence at face value, it would mean that such member zones could not be deleted via the catalog.

Just pointing out the potential problem. I have no stakes in clarifying this; probably best to find some local solution when the problem arises.

Reply:
Good point! I do feel that the name server operator's administrative control should overrule the directions within the catalog zone content.

So migrating from non-catalog-zone configured to catalog zone configures would just mean deleting the existing zone on the name server and then re-evalute the catalog zone I suppose.. Do you think we should mention this? Perhaps in the "Migrating member zones between catalogs" section?

Reply:
I'm not sure what's best, and I think it should be decided by nameserver implementors. (But deleting the zone first sounds like it would create a gap where the server is not authoritative, and zone data is lost?)

Reply:
A consumer knows what catalog zone configured what member zone. We should mention the problem, and leave it to implementers - they can, for example, add a command to make a catalog zone adopt member zones.